source/server/http_test_server_filter.cc

```diff
@@ -53,33 +53,42 @@ void HttpTestServerDecoderFilter::applyConfigToResponseHeaders(
   }
 }
 
-Envoy::Http::FilterHeadersStatus
-HttpTestServerDecoderFilter::decodeHeaders(Envoy::Http::HeaderMap& headers, bool) {
-  const auto* request_config_header = headers.get(TestServer::HeaderNames::get().TestServerConfig);
-  nighthawk::server::ResponseOptions base_config = config_->server_config();
-  absl::optional<std::string> error_message;
-
-  // TODO(oschaaf): Add functionality to clear fields
-  if (!request_config_header ||
-      mergeJsonConfig(request_config_header->value().getStringView(), base_config, error_message)) {
+void HttpTestServerDecoderFilter::sendReply() {
+  if (error_message_ == absl::nullopt) {
     decoder_callbacks_->sendLocalReply(
-        static_cast<Envoy::Http::Code>(200), std::string(base_config.response_body_size(), 'a'),
-        [this, &base_config](Envoy::Http::HeaderMap& direct_response_headers) {
-          applyConfigToResponseHeaders(direct_response_headers, base_config);
+        static_cast<Envoy::Http::Code>(200), std::string(base_config_.response_body_size(), 'a'),
+        [this](Envoy::Http::HeaderMap& direct_response_headers) {
+          applyConfigToResponseHeaders(direct_response_headers, base_config_);
         },
         absl::nullopt, "");
   } else {
     decoder_callbacks_->sendLocalReply(
         static_cast<Envoy::Http::Code>(500),
-        fmt::format("test-server didn't understand the request: {}", *error_message), nullptr,
+        fmt::format("test-server didn't understand the request: {}", *error_message_), nullptr,
         absl::nullopt, "");
   }
+}
+
+Envoy::Http::FilterHeadersStatus
+HttpTestServerDecoderFilter::decodeHeaders(Envoy::Http::HeaderMap& headers, bool done) {
```

Comment:
Nit: rename done to end_stream here and below.

Reply:
done in 8ec6e11fbac09daab718fef309f3fc122491c930 + merged master